sklearn/linear_model/_ransac.py

```diff
@@ -373,7 +373,10 @@ def fit(self, X, y, sample_weight=None):
 
             # residuals of all data for current random sample model
             y_pred = base_estimator.predict(X)
-            residuals_subset = loss_function(y, y_pred)
+            if sample_weight is None:
```

Comment:
Could you use the `sklearn.utils._check_sample_weight`. It will give something like:

```python
sample_weight = _check_sample_weigth(sample_weight, X)
residuals_subset = loss_function(y, y_pred) * sample_weight